lib/collections/teams.js

```diff
@@ -636,6 +636,67 @@ Meteor.methods({
     });
   },
 
+  // compute the score for a hypothetical puzzle response from a
+  // team. Returns the computed score. This Method does not update any
+  // database state
+  'admin.team.computeScore'(puzzle) {
+    check(puzzle, Object);
+    requireAdmin()
+    if (!Meteor.isServer) {
+      return true;
+    }
+    
+    const { puzzleId } = puzzle;
+    const masterPuzzle = Puzzles.findOne(puzzleId);
+    const endTime = puzzle.end;
+    const [tempTimeout, tempScore] = getPuzzleScoreAndTimeout(puzzle, endTime, masterPuzzle, false);
+    return [tempTimeout, tempScore];
+  },
+
+  // update the start time, end time, and hints taken information for
+  // one team and puzzle.
+  'admin.team.updatePuzzlePlay'(teamId, puzzle) {
+    check(teamId, String);
+    check(puzzle, Object);
+    requireAdmin();
+    if (!Meteor.isServer) {
+      return true;
+    }
+
+    const team = Teams.findOne(teamId);
+    if (!team) {
+      throw new Meteor.Error(400, `Oops, no team found with id ${teamId}`);
+    }
+    const { puzzleId } = puzzle;
+    const i = findIndex(team.puzzles, (p) => p.puzzleId === puzzleId);
+    if (!i) {
```

Comment:
Shouldn't this be `i > -1` or something? This means editing will fail for `puzzles[0]` if I'm not mistaken. I think this was the issue we discovered on a Zoom call way back when too but maybe I'm thinking of something else. I haven't checked this out locally to try recently